egs2/slue-voxpopuli/slu1/local/prepare_nel_data.py

```diff
@@ -0,0 +1,169 @@
+"""
+Extract entity phrase and duration tuples for evaluation
+"""
+
+import json
+import os
+import sys
+
+from datasets import load_dataset
+from fire import Fire
+from tqdm import tqdm
+
+
+def save_json(fname, dict_name):
+    with open(fname, "w") as f:
+        f.write(json.dumps(dict_name, indent=4))
+
+
+def depunctuate(text):
+    """
+    Also removes isolated apostrophe
+    """
+    text = text.replace(".", "")
+    text = text.replace("' ", " ")
+    text = text.replace("'s", " 's")
+    text = text.replace("  ", " ")
+    return text
+
+
+def read_gt_sample(gt_labels, data_idx):
+    entity_phrase = depunctuate(gt_labels[data_idx][0])
+    wrd_lst = entity_phrase.split(" ")
+    return wrd_lst, len(wrd_lst)
+
+
+def update_alignment_dct(all_word_alignments, entity_alignments, utt_id, gt_labels):
+    """
+    Update alignment_dct by appending word belonging to entity phrases with #
+    """
+    text = " ".join([item[0] for item in all_word_alignments[utt_id]])
+    text = text.replace("   ", " ")
+    text = text.replace("  ", " ")
+    if len(gt_labels) > 0:
+        data_idx, curr_idx = 0, 0
+        wrd_lst, len_phrase = read_gt_sample(gt_labels, data_idx)
+        while data_idx < len(gt_labels):  # until a match for all GT entities is found
+            label, _, _ = all_word_alignments[utt_id][curr_idx]
+            if label == wrd_lst[0]:
+                if label == "committee":
```

Comment:
remove lines 49-52 (were added for debugging)

Reply:
Fixed